service/src/main/java/greencity/service/ubs/UBSManagementServiceImpl.java

```diff
@@ -658,20 +658,22 @@ private void collectEventAboutConfirmWaste(Map<Integer, Integer> confirmed, Orde
         for (Map.Entry<Integer, Integer> entry : confirmed.entrySet()) {
             Integer capacity = bagRepository.findCapacityById(entry.getKey());
             Optional<Bag> bagOptional = bagRepository.findById(entry.getKey());
-
             if (bagOptional.isPresent() && checkOrderStatusAboutConfirmWaste(order)) {
                 Optional<Long> confirmWasteWas = Optional.empty();
+                Optional<Long> initialAmount = Optional.empty();
                 Bag bag = bagOptional.get();
                 if (Boolean.TRUE.equals(updateOrderRepository.ifRecordExist(orderId, entry.getKey().longValue()) > 0)) {
                     confirmWasteWas =
                         Optional.ofNullable(updateOrderRepository.getConfirmWaste(orderId, entry.getKey().longValue()));
+                    initialAmount =
+                        Optional.ofNullable(updateOrderRepository.getAmount(orderId, entry.getKey().longValue()));
```

Comment:
It seems like the `UpdateOrderRepository` class should be renamed because it not only updates orders

Reply:
Yes, I renamed this interface and variables accordingly.